test/boost/sstable_datafile_test.cc

```diff
@@ -4577,55 +4577,193 @@ SEASTAR_TEST_CASE(sstable_set_erase) {
 
         auto sst = sstable_for_overlapping_test(env, s, 0, key_and_token_pair[0].first, key_and_token_pair[0].first, 0);
-        set.insert(sst);
-        BOOST_REQUIRE(set.all()->size() == 1);
+        set->insert(sst);
+        BOOST_REQUIRE(set->all()->size() == 1);
 
         auto unleveled_sst = sstable_for_overlapping_test(env, s, 1, key_and_token_pair[0].first, key_and_token_pair[0].first, 0);
         auto leveled_sst = sstable_for_overlapping_test(env, s, 2, key_and_token_pair[0].first, key_and_token_pair[0].first, 1);
-        set.erase(unleveled_sst);
-        set.erase(leveled_sst);
-        BOOST_REQUIRE(set.all()->size() == 1);
-        BOOST_REQUIRE(set.all()->contains(sst));
+        set->erase(unleveled_sst);
+        set->erase(leveled_sst);
+        BOOST_REQUIRE(set->all()->size() == 1);
+        BOOST_REQUIRE(set->all()->contains(sst));
     }
 
     {
         auto cs = sstables::make_compaction_strategy(sstables::compaction_strategy_type::leveled, s->compaction_strategy_options());
-        sstable_set set = cs.make_sstable_set(s);
+        lw_shared_ptr<sstable_set> set = cs.make_sstable_set(s);
 
         // triggers use-after-free, described in #4572, by operating on interval that relies on info of a destroyed sstable object.
         {
             auto sst = sstable_for_overlapping_test(env, s, 0, key_and_token_pair[0].first, key_and_token_pair[0].first, 1);
-            set.insert(sst);
-            BOOST_REQUIRE(set.all()->size() == 1);
+            set->insert(sst);
+            BOOST_REQUIRE(set->all()->size() == 1);
         }
 
         auto sst2 = sstable_for_overlapping_test(env, s, 0, key_and_token_pair[0].first, key_and_token_pair[0].first, 1);
-        set.insert(sst2);
-        BOOST_REQUIRE(set.all()->size() == 2);
+        set->insert(sst2);
+        BOOST_REQUIRE(set->all()->size() == 2);
 
-        set.erase(sst2);
+        set->erase(sst2);
     }
 
     {
         auto cs = sstables::make_compaction_strategy(sstables::compaction_strategy_type::size_tiered, s->compaction_strategy_options());
-        sstable_set set = cs.make_sstable_set(s);
+        lw_shared_ptr<sstable_set> set = cs.make_sstable_set(s);
 
         auto sst = sstable_for_overlapping_test(env, s, 0, key_and_token_pair[0].first, key_and_token_pair[0].first, 0);
-        set.insert(sst);
-        BOOST_REQUIRE(set.all()->size() == 1);
+        set->insert(sst);
+        BOOST_REQUIRE(set->all()->size() == 1);
 
         auto sst2 = sstable_for_overlapping_test(env, s, 1, key_and_token_pair[0].first, key_and_token_pair[0].first, 0);
-        set.erase(sst2);
-        BOOST_REQUIRE(set.all()->size() == 1);
-        BOOST_REQUIRE(set.all()->contains(sst));
+        set->erase(sst2);
+        BOOST_REQUIRE(set->all()->size() == 1);
+        BOOST_REQUIRE(set->all()->contains(sst));
     }
 
     return make_ready_future<>();
 }
 
+SEASTAR_TEST_CASE(versioned_sstable_set_test) {
```

Comment:
We can split off sstable_set tests into their own file (but either before or after).

Reply:
They have been moved in following versions